Update on room and room manager

Tlhoo opened the pull request:
Methods and Javadocs added, preliminary testing required to find out what is missing, and improve methods in room manager.

Tlhoo committed:
Update on room and room manager

Methods and Javadocs added, preliminary testing required to find out what is missing, and improve methods in room manager.

stephwongwt commented on src/main/room/main/G1HRPS/Room.java:
```diff
@@ -1,33 +1,122 @@
 public class Room {
 
-	private int room_num_;
+	private String room_num_;
```
this should be int e.g. 1207, 305, 101

stephwongwt commented on src/main/room/main/G1HRPS/Room.java:
```diff
@@ -1,33 +1,122 @@
+	/**
+	 * Prints out general information of the room
+	 */
+	public void viewRoomDetails(){
```
this would be your toString. except you don't print it out. just add it all into one string with \n for each newline.

stephwongwt commented on src/main/room/main/G1HRPS/Room.java:
```diff
@@ -1,33 +1,122 @@
+	/**
+	 * Prints out whether a room is available or not
+	 */
+	public void checkRoomAvailability() {
```
Following the convention of all the other functions, the naming should be CheckRoomAvailability(), same for the rest.

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
+ */
+public void RemoveFromList(String roomNumber) { // it will not work if we just make a new room with the same parameters, since it will be stored in diff address. Suggest taking in roomNumber
+	Room room = SearchList(roomNumber);
+	system.out.println("Room with same room number found");
```
don't need to print from here. also technically when adding, should check for duplicate room number

stephwongwt replied:
Can't be resolved because this print out is not removed.

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
+ * Removes a room from room_list_, user's input is the room number
+ * @param room_num_ Room number that user wants to remove from List.
+ */
+public void RemoveFromList(String roomNumber) { // it will not work if we just make a new room with the same parameters, since it will be stored in diff address. Suggest taking in roomNumber
```
Change to room number is ok, but don't need to add the comment in code. you can comment here in the pull request to make a note.

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
+		}	
+	system.out.println("No room with details entered was found");
+	}	
+	throw new UnsupportedOperationException();
```
need to remove this line

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
+				return room_list.get(i);
+			}
+		}	
+	system.out.println("No room with details entered was found");
```
don't need to print out, same for the others, appmanager will handle output.

stephwongwt replied:
Can't be resolved because this print out is not removed.

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
+ * If same string found, returns object room
+ * @param search_text This is the input that user keys in
+ */
+public Room SearchList(String search_text) {  
```
SearchList for all managers, searches the unique identifier (in this case room number) from the list that is managed. i.e. search by room number in list of rooms.

Tlhoo replied:
In page 1 of the assignment, they wanted room to be able to search room details by both room number/guest, so I added both.

stephwongwt replied:
hm then you can do a 3 function overload.
`public Room SearchList(String search_text)`
`public Room SearchList(Guest search_guest)`
`public Room SearchList(int search_room_num)`

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
+			return room_list.get(i);
+		}
 	}
+	for(var i=0; i<room_list_.size();i++) {
```
these should be implemented directly in SearchListByRoomNum / SearchListByGuestName

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
-		throw new UnsupportedOperationException();
-	}
+/**
+ * Checks whether a room is vacant or not.(currently)
```
currently is implicit, no need to mention

stephwongwt replied:
can't be resoved, `(currently)` not removed

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
+ * Checks whether a room is vacant or not.(currently)
+ * @param roomNumber Room number of room to be checked.
+ */
+public void CheckRoomAvailabilityByRoomNumber(String roomNumber) {// consider using room type to get max number of ppl staying in the room, and compare that to guest list of room
```
Hm, what is the purpose of using room type to get max number of people staying in the room?

Tlhoo replied:
While it was not explicitly stated, the number of people staying in the same room can be diff for different room types. 
For e.g, Standard maybe for up to 2 people, VIP is for up to 4 ppl. So if max occupancy hit (2 guests in guest list of standard room), I intended for the method to reject adding any new guests to the guest list of that room. But since they didn't mention it, I'm prob overthinking and will remove that comment.

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
+public void CheckRoomAvailabilityByRoomNumber(String roomNumber) {// consider using room type to get max number of ppl staying in the room, and compare that to guest list of room
+		for(var i=0; i<room_list_.size();i++) {
+			if (room_list_.get(i).getRoomNumber()==roomNumber) {
+				room_list_.get(i).checkRoomAvailability();;
```
I thought can just return room status?

stephwongwt replied:
@Tlhoo this function needs to return something so that AppManager may use the result

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
+public void CheckRoomAvailabilityByRoomNumber(String roomNumber) {// consider using room type to get max number of ppl staying in the room, and compare that to guest list of room
+		for(var i=0; i<room_list_.size();i++) {
+			if (room_list_.get(i).getRoomNumber()==roomNumber) {
+				room_list_.get(i).checkRoomAvailability();;
```
this functions seems to be duplicated by checkRoomAvailability()

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
+		}
+}		
 
-	public List<Room> GetList() {
```
what happened to the GetList function?

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
-	public void CheckOutRoom(Room room) {
-		// TODO - implement RoomManager.CheckOutRoom
-		throw new UnsupportedOperationException();
+	throw new UnsupportedOperationException();
```
remove line

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
-	public void SetRoomStatus(Room room, RoomStatus status) {
-		// TODO - implement RoomManager.SetRoomStatus
-		throw new UnsupportedOperationException();
+	throw new UnsupportedOperationException();
```
remove line

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
+ * removes all guests elements from the guest list of a room
+ * @param roomNumber This is the room number to remove guest elements from.
+ */
+public void CheckOutAllGuest(String roomNumber) {
```
Suggest to combine CheckOutAllGuest and CheckOutOneGuest to CheckOutRoom(int room_number). You only need to clear the room's guest list and update room status.

stephwongwt replied:
This requires discussion

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
+		switch(type) { //This switch block acts as a counter for different room types
+		
+		case Single:
+			singleTotalCount+=1;
```
if you're using vs code, i suggest using Format Document to help you space out your code and make it more readable. otherwise it's fine, i can do it after combining.

Tlhoo replied:
Okay, done so with eclipse IDE formatting

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
+	suiteRoomNumber =suiteRoomNumber.substring(1,suiteRoomNumber.length()-1);
+	VIPRoomNumber =VIPRoomNumber.substring(1,VIPRoomNumber.length()-1);
+	deluxeRoomNumber =deluxeRoomNumber.substring(1,deluxeRoomNumber.length()-1);
+	System.out.println("Single: Number: "+ singleVacantCount +" out of "+singleTotalCount+" occupied");
```
Suggest to return a `HashMap<RoomType, Int[]>` which will store RoomType and an array of exactly occupied count and total count.
e.g. `{ { RoomType.Single, { 5, 10 } }, { RoomType.Suite, { 1, 5} }, { RoomType.VIP, { 0, 2} } }`

Tlhoo replied:
Sorry but I have no experience with hashmaps at all, is keeping it in the current format ok?

stephwongwt replied:
okay. but that means i'll change it afterwards. because managers should not use System.out.

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
+ * e.g. Vacant : 02-35,02-36,02-38
+ * 		Occupied:02-39,02-61
+ */
+public void GetRoomStatisticsByStatus() {
```
Suggest to return a `HashMap<RoomStatus, Int[]>` which will store RoomStatus and an array of room numbers
`e.g. { { RoomStatus.Vacant, { 401, 301, 305 } }, { RoomStatus.Occupied, { 402, 302 } }, { RoomStatus.Maintenance, { } } }`

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -1,97 +1,305 @@
+	suiteRoomNumber =suiteRoomNumber.substring(1,suiteRoomNumber.length()-1);
+	VIPRoomNumber =VIPRoomNumber.substring(1,VIPRoomNumber.length()-1);
+	deluxeRoomNumber =deluxeRoomNumber.substring(1,deluxeRoomNumber.length()-1);
+	System.out.println("Single: Number: "+ singleVacantCount +" out of "+singleTotalCount+" occupied");
```
Also just to point out, you do `System.out.println("Single: Number: "+ singleVacantCount +" out of "+singleTotalCount+" occupied");` the variable is called vacant count, but at the end of the print you say it's occupied.

Tlhoo committed:
Update for feature/room

Mainly changed formatting, and worked on  comments that steph mentioned. Also removed a few methods in room manager that isn't used in menu

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -12,18 +12,27 @@ public RoomManager() {
-	room_list_.add(room);
+	for(var i=0; i<room_list_.size();i++) {
+		if(room_list_.get(i).GetRoomNumber==room.GetRoomNumber)
+			system.out.println("Room number already exists in database.");
```
missing opening and closing brace

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -12,18 +12,27 @@ public RoomManager() {
+		else 
+			room_list_.add(room);
+		
+		
```
remove empty lines

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -35,60 +44,38 @@ public static void addNewRoom(RoomType room_type, float price, String room_numbe
  * @param room_num_ Room number that user wants to remove from List.
  */
-public void RemoveFromList(String roomNumber) { // it will not work if we just make a new room with the same parameters, since it will be stored in diff address. Suggest taking in roomNumber
+public void RemoveFromList(String roomNumber) { //will never be used by staff, remove
```
please remove comment at the end that does not mean anything for the code.

and to reply to the comment itself, this function needs to be implemented due to Supermanager

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -35,60 +44,38 @@ public static void addNewRoom(RoomType room_type, float price, String room_numbe
  * @param roomNumber Room number of room to be checked.
  */
-public void CheckRoomAvailabilityByRoomNumber(String roomNumber) {// consider using room type to get max number of ppl staying in the room, and compare that to guest list of room
+public void CheckRoomAvailabilityByRoomNumber(String roomNumber) {// prob nvr gonna be used
```
remove unneccesary comment at the end

stephwongwt commented on src/main/room/main/G1HRPS/Room.java:
```diff
@@ -43,21 +43,21 @@ public Room(RoomType room_type, float price, String room_number, BedSize bedSize
 	 * Prints out whether a room is available or not
 	 */
-	public void checkRoomAvailability() {
+	public void CheckRoomAvailability() {
```
this function is not required as manager should be the one determining this. and there's no need for printing output.

stephwongwt commented on src/main/room/main/G1HRPS/RoomManager.java:
```diff
@@ -113,9 +100,9 @@ public void SaveDB() {
  */
 public void CheckInRoom(Guest guest, String roomNumber) {//consider printing out room is occupied if user atttempts to add guest to room awlr with ppl
```
remove comment at the end of the line

Tlhoo committed:
3rd update on room and room manager

Fixed cleanliness of code, and reviewed parts mentioned.

Tlhoo committed:
Update RoomManager.java

stephwongwt committed:
Merge branch 'main' into feature/room
